tests/ui/borrowck/borrowck-overloaded-index-autoderef.stderr

```diff
@@ -17,6 +17,8 @@ LL |     let q = &mut f[&s];
    |                  ^ second mutable borrow occurs here
 LL |     p.use_mut();
    |     - first borrow later used here
+   |
+   = help: use `.split_at_mut(position)` or similar method to obtain two mutable non-overlapping sub-slices
```

Comment:
This seem to be a regression. We should check that the indexing is actually done on a slice and not a user custom type.

Reply:
I can easily check whether we have a `[]` or `Vec` and be more accurate that `split_at_mut` will work. We can also look for all methods on any custom type that being indexed on that returns `(&mut T, &mut T)` or `(&mut T, T)`. I was hoping not doing that as part of this PR, but I can.

If the comment instead means "they are doing `p.use_mut()`, where's the indexing, it happens earlier.

I am of the opinion that any type that supports indexing should have a method akin to `split_at_mut`. We should likely customize the error for types from the local crate, as *those* are much less likely to have an analogue.